gatsby/src/components/SiteDistinction/SiteDistinction.test.tsx

```diff
@@ -0,0 +1,166 @@
+				).toHaveAttribute("aria-expanded", "false");
+			});
+
+			it("should expand on click", () => {
+				render(<SiteDistinction />);
+				const button = screen.getByRole("button", {
+					name: `Show ${otherSiteTitleShort} link`,
+				});
+				userEvent.click(button);
+				expect(button).toHaveAttribute("aria-expanded", "true");
+			});
+
+			it("should change label text to hide other site link on click", () => {
+				render(<SiteDistinction />);
+				const button = screen.getByRole("button", {
+					name: `Show ${otherSiteTitleShort} link`,
+				});
+				userEvent.click(button);
+				expect(button).toHaveAttribute(
+					"aria-label",
+					`Hide ${otherSiteTitleShort} link`
+				);
+			});
+
+			it("should change data tracking attribute on button click", () => {
+				render(<SiteDistinction />);
+				const button = screen.getByRole("button", {
+					name: `Show ${otherSiteTitleShort} link`,
+				});
+				expect(button).toHaveAttribute(
+					"data-tracking",
+					`show-${otherSiteTitleShort.toLowerCase()}-link`
+				);
+				userEvent.click(button);
+				expect(button).toHaveAttribute(
+					"data-tracking",
+					`hide-${otherSiteTitleShort.toLowerCase()}-link`
+				);
+			});
+
+			it("should render an icon with an expanded class on click", () => {
+				render(<SiteDistinction />);
+				const button = screen.getByRole("button", {
+					name: `Show ${otherSiteTitleShort} link`,
+				});
+
+				// eslint-disable-next-line testing-library/no-node-access
+				const svg = button.querySelector("svg");
+				expect(svg?.classList.value).toEqual("icon");
+				userEvent.click(button);
+				expect(svg).toHaveClass("icon iconExpanded");
+			});
+
+			it("should prefetch other site link", async () => {
+				render(<SiteDistinction />);
+				const button = screen.getByRole("button", {
+					name: `Show ${otherSiteTitleShort} link`,
+				});
+				userEvent.click(button);
+
+				let prefetch: Element | null = null;
+				await waitFor(() => {
+					// eslint-disable-next-line testing-library/no-node-access
+					prefetch = document.querySelector("link[rel='prefetch']");
+					expect(prefetch).toHaveAttribute("as", "document");
+				});
+
+				expect(prefetch).toHaveAttribute(
+					"href",
+					"https://bnfc-gatsby-tests.nice.org.uk/test/?ref=switch"
+				);
+			});
+		});
+
+		describe("Link", () => {
+			it(`should have href to ${otherSiteTitleShort} with querystring parameter ref=switch`, () => {
+				render(<SiteDistinction />);
+				const link = screen.getByRole("link", {
+					name: `switch to ${otherSiteTitleShort}`,
+				});
+				expect(link).toHaveAttribute("href", expectedHref);
+			});
+
+			it(`should have data tracking attribute`, () => {
+				render(<SiteDistinction />);
+				const link = screen.getByRole("link", {
+					name: `switch to ${otherSiteTitleShort}`,
+				});
+				expect(link).toHaveAttribute(
+					"data-tracking",
+					`${otherSiteTitleShort.toLowerCase()}-link`
+				);
+			});
+		});
+
+		describe("Visually hidden", () => {
```

Comment:
There's no test for the visually hidden "currently viewing..." - I couldn't get hold of it without node access, is it worth adding one if we can?